<launch>

    <arg name="data_frequency" default="100"/>
    <arg name="namespace" value="revel"/>
    <arg name="param_file" default="control_params.yaml"/>
    
    <!-- The following controls arm compliance. Only one should be turned on at a time -->
    <arg name="dynamic_compliance" default="False"/>
    <arg name="teaching_mode" default="True"/>

    <!-- Set the joint names to be aggregated by the joint state publisher -->
    <rosparam param="joint_names">[joint_1, joint_2, joint_3, joint_4, joint_5, joint_6, finger_joint_1, finger_joint_2 ]</rosparam>

    <!-- Load the URDF/Xacro model of our robot -->
    <param name="robot_description" textfile="$(find svenzva_description)/robots/svenzva_arm.urdf" />

    <node name="joint_state_publisher" type="joint_state_publisher.py" pkg="svenzva_drivers" output="screen">
        <param name="rate" value="$(arg data_frequency)"/>
        <param name="arm_namespace" value="$(arg namespace)"/>
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_st_pub" >
    </node>

    <!-- Start the arm driver -->
    <node name="revel_driver" type="svenzva_driver.py" pkg="svenzva_drivers" output="screen">
        <param name="arm_namespace" value="$(arg namespace)"/>
        <param name="update_rate" value="$(arg data_frequency)"/>
        <param name="param_file" value="$(arg param_file)"/>
        <param name="teaching_mode" value="$(arg teaching_mode)"/>
        <param name="dynamic_compliance" value="$(arg dynamic_compliance)"/>
    </node>

    <!-- Start dynamics module-->  
    <node name="revel_dynamic_model" type="svenzva_dynamics" pkg="svenzva_drivers" output="screen"/>

</launch>
